rust/hyperlane-base/src/settings/chains.rs

```diff
@@ -108,10 +114,16 @@ impl FromRawConf<'_, RawCoreContractAddresses> for CoreContractAddresses {
                     })
                     .take_err(&mut err, path)
                     .and_then(|v| {
+                        // TODO this is sketchy
```

Comment:
More comments to explain how parsing is intended to work would be helpful

Reply:
ripped this into a better utils fn